sparse/core.py

```diff
@@ -275,10 +292,27 @@ def __getitem__(self, index):
                 i += 1
                 continue
             elif isinstance(ind, slice):
-                start = ind.start or 0
-                stop = ind.stop if ind.stop is not None else self.shape[i]
-                shape.append(min(stop, self.shape[i]) - start)
-                coords.append(self.coords[i][mask] - start)
+
+                step = ind.step if ind.step is not None else 1
+                if step > 0:
+                    start = ind.start if ind.start is not None else 0
+                    start = max(start, 0)
```

Comment:
What if someone does something like `x[-10:]` ?

Reply:
It may be useful to separate out a `normalize_slice` function that reduces slice objects to a canonical form.  It looks like there might be some useful functions in `dask/array/slicing.py` that might be helpful here.

Reply:
The `normalize_*` and `check_index` functions there might be helpful in particular

Reply:
Is there anything similar for `bool` indexing?

Reply:
On a brief look I'm not seeing much.